.github/workflows/release.yaml

```diff
@@ -0,0 +1,52 @@
+name: Release
+
+on:
+  release:
+    types: published
+
+jobs:
+  build:
+    uses: ./.github/workflows/build.yaml
+
+  release:
+    runs-on: ubuntu-latest
+    needs: build
+    permissions:
+      contents: write
+
+    steps:
+      - name: Download Artifacts
+        uses: actions/download-artifact@v4
+        with:
+          pattern: '*-release'
+          path: artifacts
+
+      - name: Release Artifact Upload
+        uses: softprops/action-gh-release@v2
+        with:
+          files: artifacts/**/*
+
+  winget:
```

Comment:
I wouldn't want this job yet in the repo, I'd suggest to drop it for now so that we can merge the rest of the CI and then once winget is in place, we can re-add it later on.